### PR TITLE
feat: add fee abs module

### PR DESCRIPTION
**Pull request by atheeshp**
closes: #775

### CODE REVIEW

File: app/ante.go [hunk 1]
```diff
@@ -51,11 +54,12 @@ func NewAnteHandler(options HandlerOptions) (sdk.AnteHandler, error) {
 	anteDecorators := []sdk.AnteDecorator{
 		ante.NewSetUpContextDecorator(), // outermost AnteDecorator. SetUpContext must be called first
 		ante.NewExtensionOptionsDecorator(options.ExtensionOptionChecker),
+		feeabsante.NewFeeAbstrationMempoolFeeDecorator(options.FeeAbskeeper),
 		ante.NewValidateBasicDecorator(),
 		ante.NewTxTimeoutHeightDecorator(),
 		ante.NewValidateMemoDecorator(options.AccountKeeper),
 		ante.NewConsumeGasForTxSizeDecorator(options.AccountKeeper),
-		cheqdante.NewDeductFeeDecorator(options.AccountKeeper, options.BankKeeper, options.FeegrantKeeper, options.TxFeeChecker),
+		feeabsante.NewFeeAbstractionDeductFeeDecorate(options.AccountKeeper, options.BankKeeper, options.FeeAbskeeper, options.FeegrantKeeper),
```
Comment (Eengineer1) [inline app/ante.go:62]: There _should_ exist a wrapping decorator that omits deducting the fee during the antehandler, performing the fee deduction process in the posthandler instead, similarly settled in the native token instead.

The differentiation here is that the chain-native token should be transferred from the `feeabs` module account to `did` module account, before it's burnt.

Let's apply the changes above + define relevant tests.

**Review by Eengineer1 — CHANGES_REQUESTED** [review]

File: app/app.go
Comment (Eengineer1) [inline app/app.go:1]: Similarly with PR #782, we do need an upgrade store record.

Please, assess + resolve.

File: go.mod [hunk 2]
```diff
@@ -9,7 +9,7 @@ require (
 	cosmossdk.io/math v1.3.0
 	filippo.io/edwards25519 v1.1.0
 	github.com/cometbft/cometbft v0.37.5
-	github.com/cometbft/cometbft-db v0.8.0
+	github.com/cometbft/cometbft-db v0.9.1
```
Comment (Eengineer1) [inline go.mod:12]: Is there any specific action required for the underlying bump here?

I'd assume upgrading to the next major version might require software changes.

If not, that's fine.

File: post/tax.go [hunk 3]
```diff
@@ -324,16 +324,37 @@ func (td *TaxDecorator) handleTaxableTransaction(
...
+	// Check if fees contain only the native denom, if it is IBC denom fee abs logic applies.
+	// The fees will be deducted from the fee payer and the tokens already sent to did module account.
+	onlyNativeDenom := true
+	for _, fee := range feeTx.GetFee() {
+		if fee.Denom != nativeDenom {
+			// If any other token besides the native denom is present, set the flag to false
+			onlyNativeDenom = false
+			break
+		}
+	}
```
Comment (Eengineer1) [inline post/tax.go:350]: This block might be better abstracted away to either a utility function or a method to the struct implementation.

e.g. `isOnlyNativeDenom`

It'll most definitely be reusable and more readable as symbolically action-based.

File: ante/fee_test.go
Comment (Eengineer1) [inline ante/fee_test.go:1]: Tests look to be sufficiently covering the intended scenarios.

**Review by Eengineer1 — CHANGES_REQUESTED** [review]
The core work looks good.

Please, address + resolve the remarks, before eventually promoting the combined work.

File: go.mod (hunk 2 above)
Comment (atheeshp) [reply go.mod:12]: fee-abs and feemarket are using this version may it is getting overridden.

**Comment by Eengineer1** [comment]
@atheeshp or @vishal-kanna Could you please resolve conflicts preparing the PR for a final review?

**Review by Eengineer1 — APPROVED** [review]
Looks well polished.

Approved.

**Pull request merged**